pkg/identity/identity.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2019, Oath Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package identity
+
+import (
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/yahoo/athenz/clients/go/zms"
+	"github.com/yahoo/athenz/libs/go/zmssvctoken"
+	"github.com/yahoo/k8s-athenz-syncer/pkg/crypto"
+	"github.com/yahoo/k8s-athenz-syncer/pkg/log"
+)
+
+var (
+	gracePeriod = 5 * time.Minute
+)
+
+// PrivateKeyProvider returns a signing key on demand.
+type PrivateKeyProvider func() (*crypto.SigningKey, error)
+
+// tokenProvider provides tokens.
+type tokenProvider interface {
+	Token() (string, error)
+}
+
+// TokenProvider implements tokenProvider provides nTokens
+type TokenProvider struct {
+	client  *zms.ZMSClient
```

Comment:
since most of the fields in `Config` struct are copied in the constructor to TokenProvider, would it make sense to have a `config Config` field in TokenProvider directly?